tap_linkedin_ads/__init__.py

```diff
@@ -21,13 +20,13 @@
 
 
 @singer.utils.handle_top_exception(LOGGER)
-def main():
+def main() -> int:
     parsed_args = singer.utils.parse_args(REQUIRED_CONFIG_KEYS)
     state = parsed_args.state or {}
-    tap(parsed_args.config, state)
+    return tap(parsed_args.config, state)
```

Comment:
this won't actually do what we want. `return 5` here does not mean `exit 5`, it just means that we are returning 5 and throwing it away.

When a python scripts raises an exception all the way to the top, it will result in an exit 1, if not, exit 0. To support custom exit codes, we need to go a bit deeper and be explicit about returning 1 / 0 first, and then introduce the 5.

Look into `sys.exit(1|2|5)`  and consider where to put it in this code.

Reply:
Maybe we want to `sys.exit()` from tap itself. However see that integrations runner is calling the tap with `sys.exit(main())`. That is why I believe that return 5 will work. 

```
CRITICAL 400 Client Error: Bad Request for url: https://www.linkedin.com/oauth/v2/accessToken
Traceback (most recent call last):
  File "/usr/local/bin/tap-linkedin-ads", line 8, in <module>
    sys.exit(main())
```

Reply:
Of course, exiting from the tap will also work. In that case, I'd rather not encapsulating the main call with sys.exit().

Reply:
but we should log as well, just so we don't just get the exit code.